test/disk_uploader_test.go

```diff
@@ -0,0 +1,134 @@
+package test
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/ptr"
+
+	v1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		imageName := os.Getenv("IN_CLUSTER_IMAGE_NAME")
+		Expect(imageName).ToNot(BeEmpty())
+
+		imageDestination := os.Getenv("IN_CLUSTER_IMAGE_REGISTRY")
+		Expect(imageDestination).ToNot(BeEmpty())
+
+		registryKeyId := os.Getenv("IN_CLUSTER_REGISTRY_ACCESS_KEY_ID")
+		Expect(registryKeyId).ToNot(BeEmpty())
+
+		registryKey := os.Getenv("IN_CLUSTER_REGISTRY_SECRET_KEY")
+		Expect(registryKey).ToNot(BeEmpty())
+
+		registryCredentials, err := createRegistryCredentials(f.CoreV1Client, f.DeployNamespace, registryKeyId, registryKey)
+		Expect(err).ToNot(HaveOccurred())
+
+		cirrosDataVolume := newCirrosDataVolume()
+		cirrosVm, err := createCirrosVM(f.KubevirtClient, f.DeployNamespace, cirrosDataVolume)
```

Comment:
please add `f.ManageVMs(cirrosVm)`, so the vm is deleted at the end of the test

Reply:
Yep sure!